ambry-api/src/main/java/com.github.ambry/clustermap/ClusterMap.java

```diff
@@ -79,4 +79,9 @@
    * Performs the required action for a replica related event.
    */
   void onReplicaEvent(ReplicaId replicaId, ReplicaEventType event);
+
+  /**
+   * Close the cluster map. Any cleanups should be done in this call.
+   */
+  void close();
```

Comment:
That is a good idea. As discussed, will have both `ClusterMap` and `ClusterParticipant` implement `AutoCloseable` and override `close()` in the respective interfaces to not throw exceptions.